src/core/lib/surface/lame_client.cc

```diff
@@ -147,18 +154,22 @@ static grpc_error *init_channel_elem(grpc_exec_ctx *exec_ctx,
 static void destroy_channel_elem(grpc_exec_ctx *exec_ctx,
                                  grpc_channel_element *elem) {}
 
-const grpc_channel_filter grpc_lame_filter = {
-    lame_start_transport_stream_op,
-    lame_start_transport_op,
-    sizeof(call_data),
-    init_call_elem,
+}  // namespace
+
+}  // namespace grpc_core
+
+extern "C" const grpc_channel_filter grpc_lame_filter = {
```

Comment:
I'm not sure the C++ filter API is lightweight enough just yet.
Better to rewrite the low level filter API to be C++ (and probably drop the
wrapper).

But this is a canary change: get a mix of C++ features into one file, and
see if it works for a release cycle before we commit.